testnet/src/node.rs

```diff
@@ -492,15 +493,25 @@ impl Node {
                                 block_header_hash: BlockHeaderHash,
                                 burn_fee: u64, 
                                 key: &RegisteredKey,
-                                parent_block: &SortitionedBlock,
-                                vrf_seed: VRFSeed) -> BlockstackOperationType {
-        
+                                burnchain_tip: &BurnchainTip,
+                                vrf_seed: VRFSeed) -> BurnchainOperationType {
+
+        let winning_tx_id = burnchain_tip.block_snapshot.winning_block_txid;
+        let mut winning_tx_vtindex = 0;
+        for op in burnchain_tip.state_transition.accepted_ops.iter() {
+            if let BlockstackOperationType::LeaderBlockCommit(op) = op {
+                if op.txid == winning_tx_id {
+                    winning_tx_vtindex = op.vtxindex
+                }
+            } 
+        }                            
```

Comment:
What happens if `winning_tx_vtindex` is never found?

Reply:
This case should not be reachable, I will add a guard condition.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1381/commits/3e4598557d541fc689d4fee396d56924371fb4c9